### PR TITLE
chore(utxo-lib): inline bitcoin-ops and pushdata-bitcoin

### PR DESCRIPTION
**Pull request by mroz22**
Second slice of #27403 — modernize `@trezor/utxo-lib` and `@trezor/address-validator` dependencies.

> **Land plan:** #28052 adds direct tests for the `pushdata.decode → null` branch and other `script` edge cases against today's `develop`. Merge #28052 first, then rebase this PR on top so the same tests guard the inlined `src/script/pushdata.ts`.

## Why

Both `bitcoin-ops` and `pushdata-bitcoin` are ~40 LOC each, unmaintained since 2017, and pulled in only by this workspace. Inlining drops two stale deps and lets us type them properly.

## Summary

- New `src/script/pushdata.ts` — TypeScript port of [`pushdata-bitcoin`](https://github.com/bitcoinjs/pushdata-bitcoin/blob/master/index.js) (3 functions: `encodingLength`, `encode`, `decode`). The return type of `decode` now correctly includes `| null` for truncated PUSHDATA payloads — the previous `declare module` signature omitted that branch even though the JS implementation always returned it.
- `src/script/ops.ts` — [`bitcoin-ops`](https://github.com/bitcoinjs/bitcoin-ops/blob/master/index.json) OPS table inlined directly, Decred extension co-located in the same file. Drops the `import * as ops from 'bitcoin-ops'` and the `declare module 'bitcoin-ops';` shim.
- `src/bufferutils.ts` — drops the local `PushDataSize` / `ReadPushDataInt` / `WritePushDataInt` aliases that only existed to compensate for missing `pushdata-bitcoin` types.
- `bufferutils.test.ts` — narrows nullability for the new `decode` return type.
- `package.json` — `bitcoin-ops` and `pushdata-bitcoin` removed.
- `scripts/list-outdated-dependencies/{connect,wallet}-dependencies.txt` — corresponding entries removed.
- `scripts/publish/babel-plugin-add-js-extension.js` — drops `bitcoin-ops` from `externalJsonImports` (only existed because the package shipped its OP table as JSON).
- `yarn.lock` deduped.

## Test status

**Direct coverage**
- [`packages/utxo-lib/tests/bufferutils.test.ts`](https://github.com/trezor/trezor-suite/blob/mroz22/issue-27403-inline-bitcoin-ops/packages/utxo-lib/tests/bufferutils.test.ts) — 14 `it` cases covering `pushDataSize` / `readPushDataInt` / `writePushDataInt` against the full fixture set (valid + invalid truncated inputs). This PR also narrows the nullability for the new `decode` return type here.
- [`packages/utxo-lib/tests/script.test.ts`](https://github.com/trezor/trezor-suite/blob/mroz22/issue-27403-inline-bitcoin-ops/packages/utxo-lib/tests/script.test.ts) — 12 `it` cases for `script.compile` / `script.decompile`, exercising the inlined OPS table and PUSHDATA decoding. Truncated-PUSHDATA cases for `OP_PUSHDATA1` / `OP_PUSHDATA2` / `OP_PUSHDATA4` are in [`__fixtures__/script.ts#L169-L177`](https://github.com/trezor/trezor-suite/blob/mroz22/issue-27403-inline-bitcoin-ops/packages/utxo-lib/tests/__fixtures__/script.ts#L169).

**Indirect coverage**
- [`packages/utxo-lib/tests/transaction.test.ts`](https://github.com/trezor/trezor-suite/blob/mroz22/issue-27403-inline-bitcoin-ops/packages/utxo-lib/tests/transaction.test.ts) — 25 cases of transaction parsing exercise OPS lookups and PUSHDATA encoding/decoding.
- [`packages/utxo-lib/tests/payments.test.ts`](https://github.com/trezor/trezor-suite/blob/mroz22/issue-27403-inline-bitcoin-ops/packages/utxo-lib/tests/payments.test.ts) — exercises script templates built on top of the OPS table.
- Aggregate: utxo-lib `test:unit` = **2451 / 2451 passing** on this branch.

**Coverage assessment:** Strong. OPS table and pushdata helpers are exercised directly by unit tests and transitively by every script-level / transaction test.

**Change safety:** **Low risk.**
- Pure TypeScript port of ~40 LOC each; logic preserved.
- Only behavioral change: `decode` return type is now correctly `… | null` for truncated payloads — the previous `declare module` signature omitted that branch even though the JS implementation always returned it.
- No new dependencies; two stale 2017-era deps dropped.

## Test plan

- [x] `yarn workspace @trezor/utxo-lib test:unit` — 2451/2451 pass
- [x] `yarn workspace @trezor/utxo-lib type-check`
- [x] `yarn workspace @trezor/utxo-lib lint:js`
- [x] `yarn workspace @trezor/utxo-lib depcheck`
- [x] `yarn workspace @trezor/connect type-check` (consumer)
- [x] `yarn workspace @trezor/blockchain-link type-check` (consumer)
- [x] `yarn dedupe`

## 🔍 Currents Test Results

🔍 **Suite desktop test results:** [View in Currents](https://app.currents.dev/projects/4ytF0E/runs?branch=mroz22/issue-27403-inline-bitcoin-ops&lookback=7)

🔍 **Suite web test results:** [View in Currents](https://app.currents.dev/projects/Og0NOQ/runs?branch=mroz22/issue-27403-inline-bitcoin-ops&lookback=7)

## 🔒 Quarantined E2E Tests

<details><summary>Trezor Suite (web) — 11 test(s)</summary>

| Test | Type |
|------|------|
| Passphrase with cardano > verify cardano address behind passphrase | 🤖 auto |
| Cardano > Basic cardano walkthrough | 🤖 auto |
| Public Keys > Check ada XPUB | 🤖 auto |
| Suite Sync - Labelling > Create new labels | 🤖 auto |
| Account types suite > Add-account-types-non-BTC-coins | 🤖 auto |
| Firmware - check readiness > Suite detects that firmware is ready | 🤖 auto |
| Export transactions > Go to account and try to export all possible variants (pdf, csv, json) | 🤖 auto |
| Quarantine test: "Onboarding - create wallet,Success (basic)" | 🙋 manual |
| Quarantine test: "Database migration,Db migration between: release/22.5/web => develop/web" | 🙋 manual |
| Quarantine test: "Multiple sessions,Session overtaken by another" | 🙋 manual |
| Quarantine test: "Recovery T2T1 - dry run,Recovery after partial recovery" | 🙋 manual |

</details>

_Updated: 2026-05-26T16:10:20.612Z • 11 test(s) total_

<details><summary>Trezor Suite (desktop) — 7 test(s)</summary>

| Test | Type |
|------|------|
| Firmware - check readiness > Suite detects that firmware is ready | 🤖 auto |
| Cardano > Basic cardano walkthrough | 🤖 auto |
| Account types suite > Add-account-types-non-BTC-coins | 🤖 auto |
| Passphrase with cardano > verify cardano address behind passphrase | 🤖 auto |
| Public Keys > Check ada XPUB | 🤖 auto |
| Onboarding - create wallet > Success (basic) | 🤖 auto |
| Quarantine test: "Multiple sessions,Session overtaken by another" | 🙋 manual |

</details>

_Updated: 2026-05-26T16:07:49.459Z • 7 test(s) total_

## 🌐 Preview deployments

🌐 **Suite Web preview:** [https://dev.suite.sldev.cz/suite-web/mroz22/issue-27403-inline-bitcoin-ops/web/](https://dev.suite.sldev.cz/suite-web/mroz22/issue-27403-inline-bitcoin-ops/web/)

### CODE REVIEW

File: packages/utxo-lib/src/script/ops.ts
Comment (mroz22) [inline packages/utxo-lib/src/script/ops.ts:1]: @copilot please output a script as a reply to this comment that fetches https://github.com/bitcoinjs/bitcoin-ops/blob/master/index.json and checks every record against OPS set and validates that all records in OPS have their counterpart in the aforementioned linked json

**Comment by mroz22** [comment]
## 🤖 PR Maintenance Journal

<details>
<summary>2026-05-20 — CI clean; Copilot review already present ✅</summary>

**CI analysis (2026-05-20 routine pass):**

All check runs: ✅ success/skipped (latest head `c40d0953`, checks from 2026-05-10/11).

No failures. No retryable runs to flag.

**Copilot review:** Already present — submitted by `copilot-pull-request-reviewer[bot]` on 2026-05-10T11:32:48Z. No action needed.

</details>

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/utxo-lib/src/script/ops.ts [hunk 1]
```diff
@@ -1,8 +1,141 @@
-import * as ops from 'bitcoin-ops';
+// Inlined from bitcoin-ops (https://github.com/bitcoinjs/bitcoin-ops),
+// unmaintained since 2017. Originally MIT-licensed.
+// Extended with Decred OP codes.
```
[the remark this answers is not in the document]
Comment (mroz22) [reply packages/utxo-lib/src/script/ops.ts]: Added full upstream MIT license header inline in c4afe0bf017.

File: packages/utxo-lib/src/script/pushdata.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// Inlined from pushdata-bitcoin (https://github.com/bitcoinjs/pushdata-bitcoin),
+// unmaintained since 2017. Originally MIT-licensed.
```
[the remark this answers is not in the document]
Comment (mroz22) [reply packages/utxo-lib/src/script/pushdata.ts]: Added full upstream MIT license header inline in c4afe0bf017.

File: packages/utxo-lib/src/script/ops.ts
Comment (mroz22) [reply packages/utxo-lib/src/script/ops.ts:1]: Parity check ran against upstream `bitcoinjs/bitcoin-ops/master/index.json`:

```
upstream count: 118
local count:    122

✅ All upstream OPS present in local with matching values

Local-only OPS (Decred stake ops, expected):
  OP_SSTX=186
  OP_SSTXCHANGE=189
  OP_SSGEN=187
  OP_SSRTX=188
```

**Review by szymonlesisz — APPROVED** [review]

**Comment by evgenysl** [comment]
QA OK
bitcoin checked
